src/coreclr/jit/lclmorph.cpp

```diff
@@ -889,10 +925,11 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
     {
         assert(val.IsAddress());
         assert(val.Node()->TypeIs(TYP_BYREF, TYP_I_IMPL));
-        assert(m_compiler->lvaVarAddrExposed(val.LclNum()));
 
         LclVarDsc* varDsc = m_compiler->lvaGetDesc(val.LclNum());
 
+        assert(m_compiler->lvaVarAddrExposed(val.LclNum()) || m_compiler->lvaVarHiddenBufferStructArg(val.LclNum()));
```

Comment:
I need to update this assert (remove `lvaVarHiddenBufferStructArg`) or revisit why we can't morph the local address despite marking it as hidden buffer.